neevaco/neeva-android
# Fix FirstRun ZeroQuery showing up all the time

fixes #1236 

### Pointers to help with review
* From Hwi: The user should never see the FirstRun ZeroQuery again after using it for the first time. 

We needed a way to see if the user has searched before on the Neeva browser to do what Hwi wants above^^. However, navigations around Neeva `settings` on the website should not count as a "search navigation". 

So I introduced a term called "Search Navigation" which is:  
- any website that is a search within the neeva domain
- any website that is outside of the neeva domain (for example, the user just typed "google.com" into the urlbar and hit enter)

What doesn't count as a "Search Navigation" is:
- anything that is not a search within the neeva domain
  - example: navigating around the neeva website and looking at settings

If the user has more than 0 Search Navigations, then that means they must have used FirstRun ZeroQuery.

## Checklists

### How was this tested?

### Manual test cases

Only "Search Navigations" (see `HistoryDao.kt` for what defines this) will set `FirstSearchDone` to true:
* Wiped app memory
* Started FirstRun
* Saw FirstRun ZeroQuery
* Navigated to Account Settings 
* Saw that ZeroQuery is still FirstRun (because a search navigation has not been made yet)
* Uses FirstRun ZeroQuery
* Sees that ZeroQuery is no longer showing the FirstRun version.


https://user-images.githubusercontent.com/20916043/219263289-0d137e1a-2dad-42b1-8e84-71db0881229f.mov


Using the FirstRun ZeroQuery from the start also hides it for the rest of the user's usage of the app:


https://user-images.githubusercontent.com/20916043/219263087-7aaa8e26-647e-4ac2-94ce-1eda208f86d6.mov



## Issues addressed
* Fixes #1236
